modules/standard/Version.chpl

```diff
@@ -22,33 +22,85 @@
-  version number plus an optional commit value.
+* :type:`versionValue`: a type that can be used to represent a semantic
+  version number plus an optional commit value. This type supports
+  `compile-time` reasoning about version numbers.
 
-* :proc:`createVersion`: a utility function for creating new version values
+* :type:`version`: a type that can be used to represent a semantic
+  version number plus an optional commit value. This type supports
+  `run-time` reasoning about version numbers.
 
-Version numbers in this module are represented using ``param`` values
+Version numbers in :type:`versionValue` are represented using ``param`` values
 to permit code specialization by being able to reason about versions
-at compile-time.
+at `compile-time`.
 
-The :type:`sourceVersion` type supports:
+The :type:`versionValue` type supports:
 
 * being printed out or cast to a ``param`` string
 
-* compile-time comparisons via ``==``, ``!=``, ``<``, ``<=``, ``>``,
+* `compile-time` comparisons via ``==``, ``!=``, ``<``, ``<=``, ``>``,
   and ``>``.  Generally speaking, "less than" corresponds to "is an
   earlier version than."  For example::
 
-    if chplVersion < createVersion(1,23) then
+    if chplVersion < new versionValue(1,23) then
       compilerWarning("This package doesn't support 'chpl' prior to 1.23.0");
 
+Version numbers in :type:`version` are represented using ``var`` values
+to permit constructing and assigning version number values at `run-time`.
+
+The :type:`version` type supports:
+
+* being printed out or cast to a string
+
+* `run-time` comparisons via ``==``, ``!=``, ``<``, ``<=``, ``>``,
+  and ``>``. Generally speaking, "less than" corresponds to "is an
+  earlier version than."
+
+The type :type:`versionValue` is useful for modifying compilation behavior,
+such as requiring a minimum version of the Chapel compiler. The type
+:type:`version` is useful for reasoning about version numbers that are not
+known at compile-time, such as values read from a text file.
+
+Comparisons between two :type:`versionValue`, two :type:`version`, or
+a :type:`versionValue` and a :type:`version` are supported.
+Equality/inequality operations check whether or not the two values
+have identical major, minor, update, and commit values.
+
+Ordered comparisons are based on the ordering of the semantic
+versions of the two values, as defined by their ``major``,
+``minor``, and ``update`` components.
+Note that ff the two values have identical semantic versions,
+any cases that rely on an ordering of the commits will generate an error
+if the values have differing, non-empty ``commit`` values due to the
+challenge of ordering commits. An empty ``commit`` value is considered to
+come after (be greater than) a non-empty value, as the former is considered
+an official release and the latter a pre-release.
+
+For example::
+
+    var v1 = new version(1, 2, 3);        // any of these can be
```

Comment:
I'm not sure what this looks like rendered, but this form of commenting a long sentence piece-by-piece at the ends of lines is unclear to me. I think it would be better to just put before this block of `vN`s, and say something like "any of the following"